packages/core/src/libraries/sso-connector.ts

```diff
@@ -107,11 +113,47 @@ export const createSsoConnectorLibrary = (queries: Queries) => {
     });
   };
 
+  /**
+   * Records the verified SAML assertion content to the database and sets the session ID to the cookie.
+   *
+   * @remarks
+   * For IdP-initiated SAML SSO flow use only.
+   * Save the SAML assertion content to the database and set the session ID to the cookie.
+   * The session ID will be used to retrieve the SAML assertion content when the user is redirected to Logto SSO authentication flow.
+   */
+  const createIdpInitiatedSamlSsoSession = async (
+    connectorId: string,
+    assertionContent: SsoSamlAssertionContent,
+    ctx: Context
+  ) => {
+    // If the assertion has a notOnOrAfter condition, we will use it as the expiration date,
+    // otherwise use the creation date + 10 minutes
+    const expiresAt = assertionContent.conditions?.notOnOrAfter
+      ? new Date(assertionContent.conditions.notOnOrAfter)
+      : new Date(Date.now() + defaultIdPInitiatedSamlSsoSessionTtl);
+
+    const { id } = await ssoConnectors.insertIdpInitiatedSamlSsoSession({
+      id: generateStandardId(),
+      connectorId,
+      assertionContent,
+      expiresAt: expiresAt.valueOf(),
+    });
+
+    // Set the session ID to cookie
+    ctx.cookies.set(idpInitiatedSamlSsoSessionCookieName, id, {
```

Comment:
nit: we can set cookies in route files to make it clear to read.

Reply:
updated, moved to the router level